src/sds.c

```diff
@@ -1021,6 +1021,22 @@ sds sdscatrepr(sds s, const char *p, size_t len) {
     return sdscatlen(s,"\"",1);
 }
 
+/* Returns one if the string contains characters to be escaped
+ * by sdscatrepr(), zero otherwise.
+ */
+int sdsneedsrepr(const sds s) {
+    size_t len = sdslen(s);
+    const char *p = s;
+
+    while (len--) {
+        if (*p == '\\' || *p == '"' || *p == '\n' || *p == '\r' ||
```

Comment:
Maybe we prefer to make that list shorter? 
Many of these are non printable, so can be handed by the last condition. 
It's right that sdscatrepr has special handling for them, so maybe it's clearer as it is (with the excessive chars)

Reply:
Except `'"'` and `'\\'` need to be check separately, others can be processed by `!isprint(*p) || isspace(*p)`.

```suggestion
        if (*p == '\\' || *p == '"' || !isprint(*p) || isspace(*p)) return 1;
```

Reply:
@chenyang8094 You are correct, but the intention here was better readability and future maintenance.